android-pdf-viewer/src/main/java/com/github/barteksc/pdfviewer/PDFView.java

```diff
@@ -462,6 +460,13 @@ public void computeScroll() {
         animationManager.computeFling();
     }
 
+    @Override
+    protected void onAttachedToWindow() {
+        super.onAttachedToWindow();
+        // Re initialise handler thread here since it's being set to null when onDetachedFromWindow is invoked
+        renderingHandlerThread = new HandlerThread("PDF renderer");

```

Comment:
@tejasna Do you think any of these HandlerThreads can get lost but keep running with no reference to them anymore? 👀 or do you they get closed/cleaned automatically?

Reply:
Yep, they'll be cleaned up automatically. There will only be one instance of the `HandlerThread` in the entire lifecycle of the view and will be cleaned up when the view is detached aka `onDetachedFromWindow`